### DISCUSSION TITLE
Maintaining package-lock/yarn.lock along with pnpm-lock

### DISCUSSION THREAD

**Discussion by vasuneet** [pnpm/pnpm#5925] [Q&A]
Hi ,

We migrated to pnpm from yarn package manager. At the last moment we stuck with SCA Agent scanning which supports only yarn.lock /package-lock and not pnpm-lock.yaml file.

We liked pnpm and dont want to revert back to yarn. So just need to maintain the yarn.lock file to  make our SCA scanning functionality.

Do we have any way to do that. I know we can generate package-lock from installed node modules but here is my doubt
does npm install -package-lock-only will work on node modules folder which is generated by pnpm ?

Or you can suggest any better way to do that.

**Comment by filipekiss** [comment]
Unfortunately this won't work. NPM is unable to read the `node_modules` generated from `pnpm` due to the way `pnpm` structures the `node_modules` folder. You will get an error similar to the one below:

```sh
$ npm install --package-lock-only
> npm ERR! Cannot read properties of null (reading 'name')
```

If your SCA Agent supports `npm-shrinkwrap.json`, you can use `npm shrinkwrap` to generate the `npm-shrinkwrap.json` file. This file behaves similarly to how `package-lock.json` would, so it's worth to give it a go. You can read more about it [here](https://docs.npmjs.com/cli/v9/commands/npm-shrinkwrap). Hope this helps. :)
